ledger/src/shred/merkle.rs

```diff
@@ -209,6 +209,25 @@ impl ShredData {
         let node = get_merkle_node(shred, SIZE_OF_SIGNATURE..proof_offset).ok()?;
         get_merkle_root(index, node, proof).ok()
     }
+
+    pub(crate) const fn const_capacity(
+        proof_size: u8,
+        chained: bool,
+        resigned: bool,
+    ) -> Result<usize, u8> {
+        debug_assert!(chained || !resigned);
+        // Merkle proof is generated and signed after coding shreds are
+        // generated. Coding shred headers cannot be erasure coded either.
+        match Self::SIZE_OF_PAYLOAD.checked_sub(
+            Self::SIZE_OF_HEADERS
+                + if chained { SIZE_OF_MERKLE_ROOT } else { 0 }
+                + (proof_size as usize) * SIZE_OF_MERKLE_PROOF_ENTRY
+                + if resigned { SIZE_OF_SIGNATURE } else { 0 },
+        ) {
```

Comment:
expect?

Reply:
expect is not const fn, would not compile